main/GarnetServer/Extensions/MyDictSet.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using Garnet.server;
+using Tsavorite.core;
+
+namespace Garnet
+{
+    public class MyDictSet : CustomObjectFunctions
+    {
+        public void AddEntry(ReadOnlySpan<byte> input, IGarnetObject garnetObject, ref (IMemoryOwner<byte>, int) output, ref RMWInfo rmwInfo)
+        {
+            Debug.Assert(garnetObject is MyDict);
+
+            int offset = 0;
+            var key = CustomCommandUtils.GetNextArg(input, ref offset).ToArray();
+            var value = CustomCommandUtils.GetNextArg(input, ref offset).ToArray();
+
+            var dictObject = (MyDict)garnetObject;
+            dictObject.TryAdd(key, value);
+            CustomCommandUtils.WriteSimpleString(ref output, "OK");
+        }
+
+        public override bool NeedInitialUpdate(ReadOnlyMemory<byte> key, ReadOnlySpan<byte> input, ref (IMemoryOwner<byte>, int) output) => true;
+
+        public override bool InitialUpdater(ReadOnlyMemory<byte> key, ReadOnlySpan<byte> input, IGarnetObject value, ref (IMemoryOwner<byte>, int) output, ref RMWInfo rmwInfo)
```

Comment:
Base class implementation of InitialUpdater could call Updater as well? So it is not needed unless user wants to do something special/different here ...